src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingOverrideCheck.java

```diff
@@ -250,36 +252,50 @@ public void visitToken(final DetailAST ast) {
 
             if (check
                 && containsTag
-                && !AnnotationUtil.containsAnnotation(ast, OVERRIDE)
-                && !AnnotationUtil.containsAnnotation(ast, FQ_OVERRIDE)) {
+                && !hasOverrideAnnotation(ast)) {
                 log(ast, MSG_KEY_ANNOTATION_MISSING_OVERRIDE);
             }
         }
     }
 
     /**
-     * Checks to see if the text block contains a inheritDoc tag.
+     * Checks to see if the ast has Override annotation.
      *
-     * @param javadoc the javadoc of the AST
-     * @return true if contains the tag
+     * @param ast AST node
+     * @return true if has Override annotation
      */
-    private static boolean containsJavadocTag(final TextBlock javadoc) {
-        boolean javadocTag = false;
-
-        if (javadoc != null) {
-            final String[] lines = javadoc.getText();
-
-            for (final String line : lines) {
-                final Matcher matchInheritDoc =
-                    MATCH_INHERIT_DOC.matcher(line);
+    private static boolean hasOverrideAnnotation(DetailAST ast) {
+        return AnnotationUtil.containsAnnotation(ast, OVERRIDE)
```

Comment:
(not related to this issue) maybe we should move this method to the `AnnotationUtil` class. It is common to check for this annotations .

Reply:
Also, this can be `containsAnnotation(ast, List.of(OVERRIDE, FQ_OVERRIDE))`

Reply:
good catch, created https://github.com/checkstyle/checkstyle/pull/11333 for it